README.md

```diff
@@ -10,13 +10,13 @@ The test data is partitioned into the following directories:
 
 - `success` - all tests in this directory should have a valid PartiQL statement that does not result in an error
 - `fail` - all tests in this directory should have a PartiQL statement that results in an error when run
+- `eval` - statements that can be evaluated and return the same result as the expected result
+- `eval-equiv` - define equivalent statements that evaluate to the same expected result
 
 Subdirectories:
 - `success/syntax` - statements that have no syntactical problems
 - `fail/syntax` - statements that have at least one syntax problem
 - `fail/static-analysis` - statements that give an error at some stage between parsing and evaluation
-- `success/eval` - statements that can be evaluated successfully and return the same result as the expected result
-- `fail/eval` - statements that throw an error during evaluation
```

Comment:
The tests in these directories were deleted since those tests were already part of the spec tests that are part of this PR.

I chose to refactor the structure of the evaluation tests because some tests tested both evaluation modes which could result in different results (i.e. success or fail). I felt like it was cleaner to compare the behaviors within the same test (and less verbose). e.g. 

```ion
    {
        name: "data type mismatch in comparison expression",
        statement: "5 > 'a'",
        assert: [
            {
                evalMode: EvalModeCoerce,
                result: EvaluationSuccess,
                output: $missing::null
            },
            {
                evalMode: EvalModeError,
                result: EvaluationFail,
            },
        ]
    },
```

I'm not too sure if there's a better directory structure I could have taken but open to other suggestions (or punting on this till we port more tests).